### PR TITLE
Enable k8s CA signing through an environmental variable

### PR DESCRIPTION
**Pull request by lei-tang**
Please provide a description for what this PR is for: 
Enable k8s CA signing through an environmental variable since the existence of "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt" does not necessarily mean using k8s CA signing. This PR is related to the issue https://github.com/istio/istio/issues/19950.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pkg/istio-agent/sds-agent.go [hunk 1]
```diff
@@ -344,7 +351,7 @@ func newSecretCache(serverOptions sds.Options) (workloadSecretCache *cache.Secre
 
 			if rootCert, err = ioutil.ReadFile(CitadelCACertPath + "/" + bootstrap.CACertNamespaceConfigMapDataName); err == nil {
 				log.Info("istiod uses self-issued certificate")
-			} else if rootCert, err = ioutil.ReadFile(k8sCAPath); err == nil {
+			} else if rootCert, err = ioutil.ReadFile(k8sCAPath); serverOptions.EnableKubernetesCa && err == nil {
```
Comment (howardjohn) [inline pkg/istio-agent/sds-agent.go]: we are checking this after we read into `rootCert`

Comment (lei-tang) [reply pkg/istio-agent/sds-agent.go]: Added "rootCert = nil" for the else case.

**Pull request merged**